cores/esp32/Arduino.h

```diff
@@ -142,7 +142,7 @@
 #endif
 #define EXTERNAL_NUM_INTERRUPTS    NUM_DIGITAL_PINS  // All GPIOs
 #define analogInputToDigitalPin(p) (((p) < NUM_ANALOG_INPUTS) ? (analogChannelToDigitalPin(p)) : -1)
-#define digitalPinToInterrupt(p)   ((((uint8_t)digitalPinToGPIONumber(p)) < NUM_DIGITAL_PINS) ? digitalPinToGPIONumber(p) : NOT_AN_INTERRUPT)
+#define digitalPinToInterrupt(p)   ((((uint8_t)digitalPinToGPIONumber(p)) < NUM_DIGITAL_PINS) ? (p) : NOT_AN_INTERRUPT)
```

Comment:
did you want to do something like this?
```
int8_t digitalPinToInterrupt(int8_t p) { 
    const int8_t pin = digitalPinToGPIONumber(p);
    return pin < NUM_DIGITAL_PINS ? pin : NOT_AN_INTERRUPT;
}
```

Reply:
What you've written is the old behavior.
The current behavior has just become a simple check to see if it is a valid pin???
I don't see yet how this is better as it doesn't convert the value.

Reply:
@TD-er true, it does not. However the vast majority of uses of `digitalPinToInterrupt` are as an input to `attachInterrupt`, which _does_ remap the pin. So this patch makes `attachInterrupt(digitalPinToInterrupt(Dx))` and `attachInterrupt(Dx)` work fine since remapping is applied once only. 
See #10367 for more context :slightly_smiling_face:

Reply:
Which begs the question, do `attachInterrupt(digitalPinToInterrupt(Dx))` and `attachInterrupt(Dx)` yield the same result? (well they do now, but probably didn't with the older code)
So is this just an optimization or can it also break existing code?

Reply:
Either has always worked on all ESP32 targets; this was reported in the above issue as broken on the Arduino Nano ESP32, the only board that _uses_ pin remapping; this PR fixed that issue restoring proper functionality everywhere.